File: c/002/main.c
/*
 * ProjectEuler 002
 * C / main.c
 * copyright (c) 2015 Susisu
 *
 * GCC / Apple LLVM version 6.0 (clang-600.0.56) (based on LLVM 3.5svn)
 * `gcc main.c -o bin/main`
 */

#include <stdio.h>

int main() {
    int sum = 0;
    int a = 0;
    int b = 1;
    while (a <= 4000000) {
        int t = a;
        a     = b;
        b    += t;
        if (a % 2 == 0) {
            sum += a;
        }
    }
    printf("%d\n", sum);
    return 0;
}

File: php/002/main.php
<?php
/*
 * ProjectEuler 002
 * PHP / main.php
 * copyright (c) 2015 Susisu
 *
 * PHP 5.4.30
 * `php main.php`
 */

function main() {
    $sum = 0;
    $a = 0;
    $b = 1;
    while ($a <= 4000000) {
        $t  = $a;
        $a  = $b;
        $b += $t;
        if ($a % 2 === 0) {
            $sum += $a;
        }
    }
    echo "$sum\n";
}

main();
